Resources/public/js/views/ez-languageselectionboxview.js

```diff
@@ -0,0 +1,402 @@
+         * @protected
+         * @param {String} eventName
+         * @param {Function} oldHandler
+         * @param {Function} newHandler
+         */
+        _syncEventHandler: function (eventName, oldHandler, newHandler) {
+            this.detach(eventName);
+            this.after(eventName, function (e) {
+                if ( newHandler ) {
+                    newHandler(e);
+                }
+                this._resetState();
+            });
+        },
+
+        /**
+         * Resets the state of the languageSelectionBox view
+         *
+         * @method _resetState
+         * @protected
+         */
+        _resetState: function () {
+            this.reset('title');
+            this.reset('confirmHandler');
+            this.reset('cancelHandler');
+            this.reset('baseTranslation');
+            this.reset('selectedNewTranslation');
+            this.reset('selectedBaseTranslation');
+            this.render();
+        },
+
+        /**
+         * Highlights the selected translation
+         *
+         * @method _uiHighlightTranslation
+         * @protected
+         * @param {String} section
+         * @param {String} languageCode
+         */
+        _uiHighlightTranslation: function (section, languageCode) {
+            this.get('container')
+                .all('.ez-' + section + '-translation')
+                .removeClass(this.TRANSLATION_HIGHLIGHT_CLASS);
+            this.get('container')
+                .one('.ez-' + section + '-translation[data-languagecode="' + languageCode + '"]')
+                .addClass(this.TRANSLATION_HIGHLIGHT_CLASS);
+        },
+
+        /**
+         * Sets the state for base translation checkbox
+         *
+         * @method _uiBaseTranslationState
+         * @protected
+         * @param {String} baseTranslation
+         */
+        _uiBaseTranslationState: function (baseTranslation) {
+            var c = this.get('container'),
+                toggledContainer = c.one('.ez-languageselectionbox-existingtranslations-container'),
+                baseCheckbox = c.one('.ez-base-translation-checkbox');
+
+            baseCheckbox.set('checked', baseTranslation);
+            if (baseTranslation) {
+                toggledContainer.addClass(this.BASE_TRANSLATION_VISIBLE_CLASS);
+            } else {
+                toggledContainer.removeClass(this.BASE_TRANSLATION_VISIBLE_CLASS);
+            }
+        },
+
+        render: function () {
+            var container = this.get('container'),
+                that = this;
+
+            container.setHTML(this.template({
+                title: this.get('title'),
+                newTranslations: this.get('newTranslations'),
+                existingTranslations: this.get('existingTranslations'),
+                canBaseTranslation: this.get('canBaseTranslation'),
+                baseTranslation: this.get('baseTranslation'),
+                selectedNewTranslation: this.get('selectedNewTranslation'),
```

Comment:
> 2. I'm passing to the template attributes: selectedNewTranslation and selectedBaseTranslation which are being set by clicking on the translation and add/remove class from translations <li>s. What I'm wodering is if I should set in template the class on <li> element that highlights selected element as probably in the future there will be need of seting those attributes on initiating the view. I wanted to do this but it needs kind of {{#if this=selectedNewStatement}} statement and AFAIK handlebars are not providing such conditions. I've seen that ppl are writing helpers for it, but I'm not sure if it's proper way or even if it's needed.

as you write those attributes are filled by the user action so don't pass them to the template as it's not needed now.